pkg/operator/azurepathfixcontroller.go

```diff
@@ -179,11 +179,11 @@ func (c *AzurePathFixController) sync() error {
 	}
 
 	azureStorage := imageRegistryConfig.Status.Storage.Azure
-	if azureStorage == nil || len(azureStorage.AccountName) == 0 {
-		return fmt.Errorf("storage account not yet provisioned")
-	}
-	if azureStorage == nil || len(azureStorage.Container) == 0 {
-		return fmt.Errorf("storage container not yet provisioned")
+	if azureStorage == nil || len(azureStorage.AccountName) == 0 || len(azureStorage.Container) == 0 {
```

Comment:
Is this going to get backported? Don't we want to ensure that the job is removed in the older versions when someone upgrades and changes the management state to removed?

Reply:
Good point... Yes, the idea is to backport this (though we might need to do it manually since the code is different on 4.14 and 4.15).
I initially put this check here as a way to avoid running the entire loop unnecessarily, but this brought more issues than it solved.
What's a good way forward? Should we simply remove the check entirely?

Reply:
Yeah, I suppose that would make sense. I don't know how big code changes you want to do (considering backporting), but we could even remove the whole `NewGeneratorAzurePathFixJob` part and most of the informers (to prevent superfluous resyncs). Our only inputs are the job name and the config.

Reply:
I like your suggestions, and I'm open to bigger improvements as long as I can understand and maintain them 😅 

When it comes to removing `NewGeneratorAzurePathFixJob`, in 4.14-4.16 we're still creating the job when it's not there, so it needs to stay, but you're right that from 4.17 onward we do not need it at all. In fact, we can even remove the generator code itself, though I'll handle that on a separate PR to simplify backporting of this one.

I'll let you know when it's ready for another pass.